File: gallery/process_counter_android/app/ui/ProcessTreePanel.kt
/** Imaginary debug panel — bind to a TextView or Compose list. */
object ProcessTreePanel {
    fun liveRootLabels(): List<String> {
        return ProcessRuntime.collectAllLiveRoots().map { proc ->
            buildString {
                append(proc.__rangerClassName)
                append(" #")
                append(proc.__rangerId)
                if (proc.__rangerPath.isNotEmpty()) {
                    append(" path=")
                    append(proc.__rangerPath)
                }
                if (proc.__rangerParentId != 0) {
                    append(" parent=")
                    append(proc.__rangerParentId)
                }
            }
        }
    }
}

File: gallery/process_counter_ios/ios/UI/ProcessTreePanel.swift
enum ProcessTreePanel {
    static func liveRootLabels() -> [String] {
        ProcessRuntime.collectAllLiveRoots().map { proc in
            var s = "\(proc.__rangerClassName) #\(proc.__rangerId)"
            if !proc.__rangerPath.isEmpty {
                s += " path=\(proc.__rangerPath)"
            }
            if proc.__rangerParentId != 0 {
                s += " parent=\(proc.__rangerParentId)"
            }
            return s
        }
    }
}
